### PR TITLE
Fix crash on POOL_release() with empty pool

### PR DESCRIPTION
**Pull request by doragasu**

### CODE REVIEW

**Review by Stephane-D — COMMENTED** [review]
I made the NULL test only in _debug_ build on purpose, just avoiding extra tests in release build as we are not supposed to use NULL for parameters

**Comment by doragasu** [comment]
Thanks for support!

In my opinion, the behavior should be the same in debug and release builds (with the exception of the log message, of course). That is, either have the check on both, or have the check on none of them.

**Long story**: I spent some hours helping a friend debug the migration of a project from 1.70 to 1.80. The game was crashing and after some time debugging I found a `MEM_alloc()` was returning NULL even though there should be plenty memory available. The weird thing is the game was working flawlessly when built with debug config, so I started searching SGDK repository history to find changes on the MEM module, on the build flags, experimenting building the library with different flags to see if one of them was causing the problem, etc. But nothing worked, that `MEM_alloc()` continued returning NULL... and I couldn't debug the debug build because it worked!

Finally I noticed the *POOL_release(): failed - pool doesn't contain any object !* log message when using the debug build, and decided to browse the code in the `pool` module. Found this and after the patch everything worked. I haven't investigated in more depth, but I think the code was calling `SPR_releaseSprite()` or some other sprite related function with no sprites available, that ended calling `POOL_release()` with an empty pool, that corrupted the MEM system (because of a NULL indirection) and caused the `MEM_alloc()` to fail.

**TL;DR**: having different behavior in debug and release builds can make debugging frustrating, because it can contribute hiding the underlying problem.

**Comment by Stephane-D** [comment]
I understand your point but adding the NULL check on release build means that i should also add that kind of _safety_ checks in many others methods (and how far should will push that ?) which can definitely start to have an impact on performance in the end. But I agree that having different behavior in _debug_ is not something we want, actually when that is the case **it is always** accompanied with a klog message telling you what is the issue and _debug_ build is really meant to be used with klog feature. What i can do is just to indeed keep same behavior and only add the klog message in debug build so we won't have different behavior with _debug_ or _release_ build (well, that is not totally true but at least that won't be the cause here)

**Comment by doragasu** [comment]
For me it's OK keeping the NULL checks only in the debug build to output the error messages, and removing the returns inside the checks so both build types behave the same. My problem was not really with the crash (I understand it is usually not a good idea calling function with NULL pointers) but with the different behavior that did not allow me use the debug build to spot the crash.

**Comment by Stephane-D** [comment]
Ok ! I will modify the code so it just log the error without interrupting process and do the same for the sprite engine which also contains that kind of debug checks.

**Comment by doragasu** [comment]
Awesome, thanks!
